app/controllers/projects_controller.rb

```diff
@@ -1,17 +1,30 @@
 class ProjectsController < ApplicationController
-  before_action :set_project, except: %i[new create index archived]
+  before_action :set_project, except: %i[new create index archived join]
 
   Project::MAX_MEMBERS_PER_CARD = 4;
 
   # GET /projects
   # GET /projects.xml
   def index
-    @projects = policy_scope(Project).not_archived
-    @activities_group = Activity.grouped_activities(@projects, 1.week.ago)
+    @projects = Hash.new
+
+    current_user_projects = policy_scope(Project)
+
+    @projects[:is_member] = Projects::ProjectsSerializer::from_collection(
+      ProjectDecorator.decorate_collection(current_user_projects)
+    )
+
+    @projects[:without_current_user] = Projects::ProjectsSerializer::from_collection(
+      ProjectDecorator.decorate_collection(
+        current_team.projects.joinable.where.not(id: current_user.projects).order(:updated_at)
```

Comment:
You're using the same query (a big one) in two different places [(here)](https://github.com/Codeminer42/cm42-central/pull/107/files#diff-757cba42ec9599bac53d4cc33cdde3bfR106).
What do you think of moving this code to some methods to make it easier to read (and to test)?

I'd suggest you some refactoring.

- First I'd break the query into some methods:
```Ruby
private

def projects_to_join
  @projects_to_join ||= team_projects.joinable.where.not(id: current_user.projects)
end

# Maybe move those methods to ApplicationController, as current_team is there...
def team_projects
  @team_projects ||= current_team.projects
end

def user_projects
  @user_projects ||= current_user.projects
end
```

Also, maybe it's better to use scopes instead of making database call straight from controller.

- Then I'd break the query into some scopes:

In **Project.rb**:
```Ruby
scope :by_slug, -> (slug) { where(slug: slug) }

scope :joinable_except, -> (project_ids) { joinable.where.not(id: project_ids) }

scope :asc_by_updated_at, -> { order(:updated_at, :asc) }
```

Then you could just update your `ProjectsController.rb` code to:
```Ruby
def projects_to_join
  team_projects.joinable_except(user_projects.pluck(:id))
end
```
Then, at this line, you just need to call:
```Ruby
Projects::ProjectsSerializer::from_collection(
    ProjectDecorator.decorate_collection(
      projects_to_join.asc_by_updated_at
    )
)
```

And in [this line](https://github.com/Codeminer42/cm42-central/pull/107/files#diff-757cba42ec9599bac53d4cc33cdde3bfR106), you could just call:
```Ruby
project = projects_to_join.by_slug(params[:id]).first
```